ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
Which places are you thinking of? The two `ssl_get_new_session` calls I see aren't error cases but cases where we decide not to reuse a session. This one is a straight-up error path, most analogous to the `SSL_R_OLD_SESSION_CIPHER_NOT_RETURNED` case which doesn't reset `s->session`.